sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -419,66 +513,98 @@ struct tpu_params<tpu::amx, Ta, Tb, Tc, 0, 0, 0,
+
+```
+==== Runtime Query
+This provides a more general query interface with information about
+sizes, types,  and scopes that are supported by a specific matrix
+implementation. This is needed to avoid padding by the user, for
+tuning, and efficient code generation if used by a library. The
+general query returns an array of `combinations` of `combination`
+type. Each combination includes the sizes and the types for the
+matrices A, B, and accumulator. Note that for each matrix hardware,
+the query returns `max_msize, max_nsize, max_ksize` or `msize, nsize,
+ksize` exclusively, depending on whether the implementation supports a
+continuous or discrete number of sizes. For example, the Intel AMX
+implementation supports a continuous number of sizes, so the `max_*`
+variant is applied and only the maximum number is returned. The Intel
+XMX implementation, on the other hand, supports a discrete list of
+numbers so the `msize, nsize, ksize` variant is applied.
+
+The table below provides a description for each of the device matrix
+desciptors that can be queried using `get_info` API.
+
+[frame="none",options="header"]
+|======================
+| Device descriptors | Return type| Description
+|`ext::oneapi::experimental::info::device::matrix::numtiles`| `uint32_t`
+|indicates number of tiles in Intel AMX (does not apply to Intel XMX)
+|`ext::oneapi::experimental::info::device::matrix::scopes`
+|`std::vector<scope_t>`
+|indicates the memory and execution scopes supported by the matrix
+implementation
+|`ext::oneapi::experimental::info::device::matrix::num_scopes`|`uint32_t`
+|indicates number of scopes supported by the matrix implementation
+|`ext::oneapi::experimental::info::device::matrix::combinations` |
+`std::vector<combination>`| tells the set of supported matrix sizes
+and types
+|`ext::oneapi::experimental::info::device::matrix::num_combinations`|`uint32_t`
+|indicates number of combinations supported by the matrix
+implementation which corresponds to the size of the `combinations` array
+|`combination` 
+|`combination`
+|composes the types and sizes of A, B, accumulator matrices allowed in
+one combination
+|`max_msize`, `max_nsize`, `max_ksize`| `uint32_t`| if the matrix
+implementation supports a continuous number of element sizes, each of
+these members is non-zero, and the matrix implementation supports all
+element sizes from 1 up to (and including) that number. By contrast,
+if the TPU implementation supports a discrete number of element
+sizes, each of these members has the value zero
+|`msize`, `nsize`, `ksize`| `uint32_t`| if the matrix implementation
+supports a discrete number of element sizes, each of these members is
+non-zero, and the value tells one of the supported element sizes. By
+contrast, if the matrix hardware supports a continuous number of
+element sizes, each of these members has the value zero
+|`atype`, `btype`, `accumulatortype`| `matrix_type` | indicates the
+types supported in the combination
```

Comment:
This would be a good place to describe what each of the `matrix_type` enumerators mean.  For example:

> Tells the combination of types that are supported for the _A_, _B_, and _accumulator_ matrices in the `T` template parameter as follows:
>
> * `bf16`: `sycl::bfloat16`
> * `fp16`: `sycl::half`
> * `tf32`: `sycl::ext::oneapi::experimental::matrix::precision::tf32`
> * `fp32`: `float`
> ...